src/mtcp/mtcp_restart.c

```diff
@@ -239,9 +239,11 @@ MTCP_PRINTF("Attach for debugging.");
       mtcp_abort();
     }
     // This assumes that the MTCP header signature is unique.
-    do {
-      mtcp_readfile(rinfo.fd, &mtcpHdr, sizeof mtcpHdr);
-    } while (mtcp_strcmp(mtcpHdr.signature, MTCP_SIGNATURE) != 0);
+    mtcp_readfile(rinfo.fd, &mtcpHdr, sizeof mtcpHdr);
```

Comment:
I was writing too fast, while distracted earlier.  There was only one
patch (mine).  In any case,  I hope my explanation was clear.

On Sat, Mar 21, 2015 at 01:59:10PM -0400, Gene Cooperman wrote:
> Hi Kapil,
>> THIS NEXT LINE WAS MY MISTAKE.  I APOLOGIZE.
>>     I don't understand your modification of my patch.
>     In the current code, mtcp_readfile() either succeeds in reading the
> header, or it calls mtcp_abort().  The existing do/while loop says
> that if we succeeded in reading the header (e.g., 4096 bytes) and if it
> does not match MTCP_SIGNATURE, then call mtcp_readfile() another time
> and read the next 4096 bytes and see if you can find an MTCP_SIGNATURE
> there.  After repeatedly failing to find MTCP_SIGNATURE,
> mtcp_readfile() is called when the fd is at the end of the ckpt image,
> and it reports the obscure error mesage
> " failed to read after 10 tries in a row.\n", and then aborts.  :-)
>     I patched the original behavior to call mtcp_readfile() only
> once, and then abort with a relevant error message if the MTCP_SIGNATURE
> doesn't match.
>> THESE THREE LINES BELOW ARE MY MISTAKE.  I APOLOGIZE:
>>  In your newest patch, you seem to be proposing that if
>> it doesn't match with MTCP_SIGNATURE, then don't abort at all, and try
>> to process it as an MTCP header even though it's not one.  :-(
> 
> Best,
> - Gene
> 
> On Sat, Mar 21, 2015 at 09:02:47AM -0700, Kapil Arya wrote:
> > > @@ -239,9 +239,11 @@ MTCP_PRINTF("Attach for debugging.");
> > >        mtcp_abort();
> > >      }
> > >      // This assumes that the MTCP header signature is unique.
> > > -    do {
> > > -      mtcp_readfile(rinfo.fd, &mtcpHdr, sizeof mtcpHdr);
> > > -    } while (mtcp_strcmp(mtcpHdr.signature, MTCP_SIGNATURE) != 0);
> > > +    mtcp_readfile(rinfo.fd, &mtcpHdr, sizeof mtcpHdr);
>> AND THIS LINE BELOW WAS MY MISTAKE.
>> [ Above line is Kapil's proposed alternative patch.  - Gene ]